app/services/CheckoutService.scala

```diff
@@ -276,10 +278,10 @@ class CheckoutService(identityService: IdentityService,
       val newRatePlan = RatePlan(renewal.plan.id.get, None)
       val renewCommand = Renew(subscription.id.get, subscription.termStartDate, subscription.termEndDate, NonEmptyList(newRatePlan), contractEffective, customerAcceptance)
       val validPromotion = for {
-        code <- renewal.promoCode
-        deliveryCountryString <- contact.mailingCountry
-        deliveryCountry <- CountryGroup.countryByName(deliveryCountryString)
-        validPromotion <- promoService.validate[NewUsers](code, deliveryCountry, renewal.plan.id).toOption // would be better to have valid/invalid/tracking
+        code <- renewal.promoCode.withLogging("promo code from client")
+        deliveryCountryString <- contact.mailingCountry.withLogging("salesforce mailing country")
+        deliveryCountry <- CountryGroup.countryByName(deliveryCountryString).withLogging("delivery country object")
+        validPromotion <- promoService.validate[com.gu.memsub.promo.Renewal](code, deliveryCountry, renewal.plan.id).withLogging("validated promotion").toOption
```

Comment:
this will look like this in the logs, any thoughts on the format? (and I think no PII!) @jacobwinch @paulbrown1982 :
```
2017-01-20 14:31:44,394 com.gu.memsub.promo.LogImplicit$[LogImplicit.scala:9] INFO: promo code from client {Some(PromoCode(JOHN2017))}
2017-01-20 14:31:44,396 com.gu.memsub.promo.LogImplicit$[LogImplicit.scala:9] INFO: salesforce mailing country {Some(United Kingdom)}
2017-01-20 14:31:44,397 com.gu.memsub.promo.LogImplicit$[LogImplicit.scala:9] INFO: delivery country object {Some(Country(GB,United Kingdom,List()))}
2017-01-20 14:31:44,401 com.gu.memsub.promo.LogImplicit$[LogImplicit.scala:9] INFO: found promo {\/-(Promotion(1a2ebbd7-1033-4a94-9e0b-ce5612a4c556,JOHN2017,JOHN2017,AppliesTo(Set(ProductRatePlanId(2c92c0f8574b2b8101574c4a9480068d), ProductRatePlanId(2c92c0f8574b2be601574c39888d6850), ProductRatePlanId(2c92c0f858aa38af0158da325d2f0b3d)),Set(Country(GB,United Kingdom,List()))),CampaignCode(C_ITOK4F3N),Map(Channel(Channel 0) -> Set(PromoCode(JOHN2017))),None,2017-01-04T00:00:00.000Z,None,PercentDiscount))}
2017-01-20 14:31:44,401 com.gu.memsub.promo.LogImplicit$[LogImplicit.scala:9] INFO: upgrade/renewal for context {\/-(PercentDiscount)}
2017-01-20 14:31:44,403 com.gu.memsub.promo.LogImplicit$[LogImplicit.scala:9] INFO: effective promotion type {\/-(Tracking$)}
2017-01-20 14:31:44,404 com.gu.memsub.promo.LogImplicit$[LogImplicit.scala:9] INFO: validated promo type {\/-(Tracking$)}
2017-01-20 14:31:44,405 com.gu.memsub.promo.LogImplicit$[LogImplicit.scala:9] INFO: validated promotion {\/-(ValidPromotion(PromoCode(JOHN2017),Promotion(1a2ebbd7-1033-4a94-9e0b-ce5612a4c556,JOHN2017,JOHN2017,AppliesTo(Set(ProductRatePlanId(2c92c0f8574b2b8101574c4a9480068d), ProductRatePlanId(2c92c0f8574b2be601574c39888d6850), ProductRatePlanId(2c92c0f858aa38af0158da325d2f0b3d)),Set(Country(GB,United Kingdom,List()))),CampaignCode(C_ITOK4F3N),Map(Channel(Channel 0) -> Set(PromoCode(JOHN2017))),None,2017-01-04T00:00:00.000Z,None,Tracking$)))}

```